src/App/Fossa/Analyze.hs

```diff
@@ -53,77 +52,63 @@ import qualified Strategy.Ruby.BundleShow as BundleShow
 import qualified Strategy.Ruby.GemfileLock as GemfileLock
 import Types
 
-data ScanCmdOpts = ScanCmdOpts
-  { cmdBasedir :: FilePath
-  , cmdDebug   :: Bool
-  , cmdOutFile :: Maybe FilePath
-  , cmdApiKey  :: Maybe Text
-  } deriving (Eq, Ord, Show, Generic)
-
-scanMain :: ScanCmdOpts -> IO ()
-scanMain ScanCmdOpts{..} = do
-  hSetBuffering stdout NoBuffering
-  hSetBuffering stderr NoBuffering
-  basedir <- validateDir cmdBasedir
-
-  scan basedir cmdOutFile cmdApiKey
-    & withLogger (bool SevInfo SevDebug cmdDebug)
-    & runThreaded
-
-validateDir :: FilePath -> IO (Path Abs Dir)
-validateDir dir = do
-  absolute <- resolveDir' dir
-  exists <- doesDirExist absolute
-
-  unless exists (die $ "ERROR: Directory " <> show absolute <> " does not exist")
-
-  pure absolute
-
-scan ::
+data ScanDestination
+  = UploadScan (Url 'Https) Text ProjectMetadata -- ^ upload to fossa with provided api key and base url
+  | OutputStdout
+  deriving (Eq, Ord, Show, Generic)
+ 
+analyzeMain :: Severity -> ScanDestination -> Maybe Text -> Maybe Text -> IO ()
+analyzeMain logSeverity destination name revision = do
+  basedir <- getCurrentDir
+  runThreaded $ withLogger logSeverity $ analyze basedir destination name revision
+
+analyze ::
   ( Has (Lift IO) sig m
   , Has Logger sig m
   , Has Threaded sig m
   , MonadIO m
   , Effect sig
   )
   => Path Abs Dir
-  -> Maybe FilePath
-  -> Maybe Text -- ^ API key for fossa core. when present, we'll upload results
+  -> ScanDestination
+  -> Maybe Text -- ^ cli override for name
```

Comment:
This is gross: we should pass in the exact `ProjectRevision` here -- will require a future refactor to move project inference outside of analyze/test.

currently blocked on a usable `Async` implementation for fused-effects